### PR TITLE
Added documentation for options parameter in ConnectionConfiguration

### PR DESCRIPTION
**Pull request by killaitis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,10 +45,11 @@ Publisher<? extends Connection> connectionPublisher = connectionFactory.create()
 ConnectionFactory connectionFactory = ConnectionFactories.get(ConnectionFactoryOptions.builder()
    .option(DRIVER, "postgresql")
    .option(HOST, "...")
-   .option(PORT, "...")  // optional, defaults to 5432
+   .option(PORT, 5432)  // optional, defaults to 5432
    .option(USER, "...")
    .option(PASSWORD, "...")
    .option(DATABASE, "...")  // optional
+   .option(OPTIONS, Map.of("lock_timeout", "30s", "statement_timeout", "5m")) // optional
```
Comment (mp911de) [inline README.md]: Let's not use Java 9+ syntax here as this sample might leave readers with the assumption that they require a newer Java version to run this code. All R2DBC components use Java 8 as baseline.

File: README.md [hunk 2]
```diff
@@ -83,10 +85,11 @@ Mono<Connection> connectionMono = Mono.from(connectionFactory.create());
 ```java
 ConnectionFactory connectionFactory = new PostgresqlConnectionFactory(PostgresqlConnectionConfiguration.builder()
     .host("...")
-    .port("...").  // optional, defaults to 5432
+    .port(5432)  // optional, defaults to 5432
     .username("...")
     .password("...")
     .database("...")  // optional
+    .options(Map.of("lock_timeout", "10s", "search_path", "public, myschema")) // optional
```
Comment (mp911de) [inline README.md]: Same here.

File: README.md (hunk 1 above)
Comment (killaitis) [reply README.md]: Yeah, but creating a map in Java 8 is sooo ugly and bloats the code, so I chose Java 9 style. I've changed it to J8.

File: README.md (hunk 2 above)
Comment (killaitis) [reply README.md]: also moved to J8 style.

**Comment by mp911de** [comment]
Thanks a lot, that's merged now.

**Pull request merged**